frontend/src/components/SettingsModal.jsx

```
@@ -299,467 +299,329 @@ const SettingsModal = ({ isOpen, onClose, onSave, activeModel }) => {
-                                    >
-                                        <div className="w-2 h-2 rounded-full bg-green-500 animate-pulse" />
-                                        Recently Indexed
-                                        {showHistory ? <ChevronUp className="w-3 h-3" /> : <ChevronDown className="w-3 h-3" />}
-                                    </button>
+                            Settings
+                        </h2>
+                        <button
+                            onClick={onClose}
+                            className="p-2 rounded-xl hover:bg-secondary transition-colors"
+                            aria-label="Close settings"
+                        >
```

Biome (2.4.6)

[error] Provide an explicit type prop for the button element.

(lint/a11y/useButtonType)